### ISSUE TITLE
e2e flake: failed to create 3 members etcd cluster: still failing after 5 retries

### ISSUE THREAD

**Issue by hongchaodeng** [coreos/etcd-operator#581]
From the logs, it looked like flaky:
```
time="2017-01-03T23:02:27Z" level=info msg="added member (test-etcd-gzogq-0002)" cluster-name=test-etcd-gzogq pkg=cluster 
time="2017-01-03T23:02:32Z" level=info msg="skip reconciliation: running ([test-etcd-gzogq-0000 test-etcd-gzogq-0001]), pending ([test-etcd-gzogq-0002])"
time="2017-01-03T23:02:35Z" level=info msg="deleting cluster" cluster-name=test-etcd-gzogq pkg=cluster 
```

**Comment by hongchaodeng** [comment]
Happened again:

This time it's a little different:
- it successfully created the first cluster, deleted the cluster, and restored the seed.
- it failed to create the third member of the second cluster (restored)
```
restore_test.go:126: failed to create 3 members etcd cluster: still failing after 18 retries
```

**Comment by hongchaodeng** [comment]
I looked at the logs. The whole thing seems to take very long.

This is where the first cluster was deleted
```
time="2017-01-06T17:59:27Z" level=info msg="deleting cluster" cluster-name=test-etcd-selbd pkg=cluster 
```

This is where the second cluster was failed and deleted
```
time="2017-01-06T18:02:28Z" level=info msg="deleting cluster" cluster-name=test-etcd-selbd pkg=cluster 
```
It roughly matches 3 mins = 3*60s = 18 (retry) * 10s.

The cluster was starting to restore at
```
time="2017-01-06T18:00:57Z" level=info msg="restoring cluster from existing backup (test-etcd-selbd)" cluster-name=test-etcd-selbd pkg=cluster 
```

That's already 90s passed! Then nothing seems to be wrong but no enough time left!

**Comment by hongchaodeng** [comment]
Conclusion:
- We need to add timestamp to e2e test log
- We need to measure how long each operation takes, have better expectation, and understanding distribution

**Comment by hongchaodeng** [comment]
Cause:
- The reason why it took so long between deleting previous cluster and receiving a new cluster to restore is that our test is concurrent but operator is blocking. In our test, we have two restore cases to do, each creation of seed member will take 90s. This will likely cause another to fail.

**Comment by hongchaodeng** [comment]
I actually think this should be the issue of #588. Moving there.

**Comment by hongchaodeng** [comment]
This is similar to #588 
fixed via #639
Never seen this after we fix the bug.
